### DISCUSSION TITLE
TileLayer Force getTileData With updateTriggers?

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#9039] [Q&A] [answered]
Hi! Do updateTriggers work with the TileLayer getTileData function?

I am attempting to trigger the getTileData function on an interval basis using an updateTriggers variable. However, I never see network calls to retrieve tiles when the updateTrigger variable changes...

Below is a simple CodePen where you can see the behavior...

[https://codepen.io/jfuehner/pen/BagzVjr](https://codepen.io/jfuehner/pen/BagzVjr)

I _am_ able to get this to work if I call the TileLayer's updateState function directly passing in an _params_ object

```
{
   changeFlags: {
      updateTriggersChanged: {
          all: true
      }
   }
}
```

This feels like a hack though...

**Comment by Pessimistress** [comment, marked as answer]
You are changing the `updateTileData` variable, not the `updateTriggers` prop. 

[Update layers](https://deck.gl/docs/developer-guide/using-layers#updating-layers)

**Comment by jfuehner** [comment]
I see… Thanks for clearing up my misunderstanding of updateTriggers. 

My interpretation was changing a value assigned to an updateTrigger property would force the layer assessor or layer function to fire re-rendering that portion of the layer automatically… I didn’t realize I would need to set decks array of layers again.
